src/os/bluestore/BlueStore.cc

```diff
@@ -6964,7 +6964,7 @@ void BlueStore::_do_write_small(
   if (ep != o->extent_map.extent_map.begin()) {
     --ep;
     b = ep->blob;
-    if (ep->logical_offset + b->get_blob().get_ondisk_length() <= offset) {
+    if ((ep->logical_offset - ep->blob_offset) + b->get_blob().get_ondisk_length() <= offset) {
```

Comment:
I ran ceph_test_objectstore(only BlueStore), here is the result.

[----------] Global test environment tear-down
[==========] 61 tests from 1 test case ran. (9295011 ms total)
[  PASSED  ] 61 tests.